### PR TITLE
merge task system into operation system and make it compatible with the action system.

### PR DESCRIPTION
**Pull request by wangmingliang-ms**

### CODE REVIEW

File: azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/appservice/function/core/AzureFunctionPackager.java [hunk 1]
```diff
@@ -240,7 +240,7 @@ private void copyJarsToStageDirectory(FunctionProject project) throws IOExceptio
     }
 
     private void trackFunctionProperties(Map<String, FunctionConfiguration> configMap) {
-        AzureTelemetry.getActionContext().setProperty(TRIGGER_TYPE, StringUtils.join(getFunctionBindingList(configMap), ","));
+        AzureTelemetry.getContext().getActionParent().setProperty(TRIGGER_TYPE, StringUtils.join(getFunctionBindingList(configMap), ","));
```
Comment (Flanker32) [inline azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/appservice/function/core/AzureFunctionPackager.java:243]: `getActionParent` or `getParentAction` or `getCurrentAction`?

File: azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/Executable.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+package com.microsoft.azure.toolkit.lib.common;
+
+@FunctionalInterface
+public interface Executable<T> {
```
Comment (Flanker32) [inline azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/Executable.java:9]: Callable<T>?

File: azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/appservice/function/core/AzureFunctionPackager.java (hunk 1 above)
Comment (wangmingliang-ms) [reply azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/appservice/function/core/AzureFunctionPackager.java:243]: get the parent(actually ancestor) context whose type is `ACTION`

**Review by Flanker32 — APPROVED** [review]
LGTM

**Pull request merged**